kubernetes/app/external-secret.yaml

```
@@ -0,0 +1,29 @@
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: crud-auth-external-secrets
+  namespace: crud-app
+spec:
+  refreshInterval: 30m          # rate SecretManager pulls GCPSM
+  secretStoreRef:
+    kind: ClusterSecretStore
+    name: gcp-secret-store-dev     # name of the SecretStore (or kind specified)
```

```suggestion
    name: gcp-secret-store    # name of the SecretStore (or kind specified)
```

In `kubernetes/external-secrets/secret-store.yaml`
You've specified the name as `gcp-secret-store` should either change the name on that file or on here so the reference to the ClusterSecretStore works :)